packages/core-flows/src/definition/cart/utils/prepare-confirm-inventory-input.ts

```diff
@@ -0,0 +1,61 @@
+import { ProductVariantDTO } from "@medusajs/types"
+import { MedusaError } from "medusa-core-utils"
+
+interface ConfirmInventoryPreparationInput {
+  productVariantInventoryItems: {
+    variant_id: string
+    inventory_item_id: string
+    required_quantity: number
+  }[]
+  items: { variant_id?: string; quantity: number }[]
+  variants: ProductVariantDTO[]
+  locationIds: string[]
+}
+
+export const prepareConfirmInventoryInput = ({
+  productVariantInventoryItems,
+  locationIds,
+  items,
+  variants,
+}: ConfirmInventoryPreparationInput) => {
+  if (!productVariantInventoryItems.length) {
+    return []
+  }
+
+  const variantsMap = new Map<string, ProductVariantDTO>(
+    variants.map((v) => [v.id, v])
+  )
+
```

Comment:
suggestion: you could create a map of productVariantInventoryItems in order to not have to do a find for each item later. following what you did with the variant map